.github/workflows/NodeJS.yml

```diff
@@ -115,15 +115,77 @@ jobs:
         shell: bash
         run: ./scripts/node_build.sh ${{ matrix.node }}
 
-  osx-nodejs:
-    if: startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/main' || github.repository != 'duckdb/duckdb-node'
+  osx-nodejs-arm64:
     name: node.js OSX
-    runs-on: macos-latest
+    runs-on: macos-14
     needs: set-up-npm
     continue-on-error: ${{ matrix.node != '18' && matrix.node != '20' && matrix.node != '21' }}
     strategy:
       matrix:
-        target_arch: [ x64, arm64 ]
+        target_arch: [ arm64 ]
+        node: [ '16', '17', '18', '19', '20', '21', '22']
+        isRelease:
+          - ${{ startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/main' }}
+        exclude:
+          - isRelease: false
+            node: 12
+          - isRelease: false
+            node: 14
+          - isRelease: false
+            node: 16
+          - isRelease: false
+            node: 17
+          - isRelease: false
+            node: 19
+          - target_arch: arm64
+            node: 12
+          - target_arch: arm64
+            node: 14
+        # these older versions of NodeJS don't have M1 support
+
+    env:
+      TARGET_ARCH: ${{ matrix.target_arch }}
+      DUCKDB_NODE_BUILD_CACHE: 0
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      # Default Python (3.12) doesn't have support for distutils
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Setup Ccache
+        uses: hendrikmuhs/ccache-action@main
+        with:
+          key: ${{ github.job }}-${{ matrix.target_arch }}
+          save: ${{ ( github.ref == 'refs/heads/main' || github.repository != 'duckdb/duckdb-node' ) && ( matrix.node == '19' ) }}
+
+      - name: Downgrade curl # fixes a bug with the brew curl that lead to failed downloads
+        shell: bash
+        run: |
+          brew uninstall --ignore-dependencies curl
+          which curl
+
+      - name: Setup
+        shell: bash
+        run: ./scripts/node_version.sh
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NODE_AUTH_TOKEN}}
+
+      - name: Node ${{ matrix.node }}
+        shell: bash
+        run: ./scripts/node_build.sh ${{ matrix.node }}
+
+  osx-nodejs-x64:
+    name: node.js OSX
+    runs-on: macos-13
+    needs: set-up-npm
+    continue-on-error: ${{ matrix.node != '18' && matrix.node != '20' && matrix.node != '21' }}
+    strategy:
+      matrix:
+        target_arch: [ x64 ]
```

Comment:
Is it necessary to exclude those node versions when building for x86, if the reason for excluding them is that they are not supported on arm?

Reply:
Which combination of version / platform are you referring to?

Reply:
I apologize if I am misreading the code.
As I understand it, the job `osx-nodejs-x64` is excluding some versions, because they are not supported on ARM, given the comment `these older versions of NodeJS don't have M1 support`.
![image](https://github.com/duckdb/duckdb-node/assets/23435481/573d8ff9-0b22-43d1-8390-e15fc6e3322d)

Reply:
Right, currently there is quite a bit of duplication, where osx-nodejs-arm64 and osx-nodejs-x64 are basically the same. I think for now it's sort of OK, and makes it a bit more clear (at least to me). I will then move to have an outer matrix that selects the runner / platform (so unifying those 2 blocks together).